tests/unit/lib/test_opensearch_lock.py

```diff
@@ -0,0 +1,200 @@
+        mock_response_nodes(self.charm.unit.name, self.charm.opensearch.host)
+        """Initially no one has the lock if there is a leader"""
+        self.harness.set_leader(is_leader=True)
+
+        assert self.harness.get_relation_data(self.lock_rel_id, self.charm.app.name) == {}
+        assert self.harness.get_relation_data(self.lock_rel_id, self.charm.unit.name) == {}
+        assert self.harness.get_relation_data(self.lock_rel_id, f"{self.charm.app.name}/0") == {}
+
+    def test_node_lock_no_online_hosts_leader_acquire_lock_via_databag(self):
+        # The leader does not have the lock at this moment, but starts "queuing up"
+        # If the charm code raises an uncaught exception later in the Juju event
+        self.harness.set_leader(is_leader=True)
+        self.harness.update_relation_data(
+            self.lock_rel_id, f"{self.charm.app.name}", {"unit-with-lock": ""}
+        )
+        assert not self.harness.charm.node_lock.acquired
+
+        assert (
+            self.harness.get_relation_data(self.lock_rel_id, f"{self.charm.unit.name}")[
+                "lock-requested"
+            ]
+            == "true"
+        )
+
+        # A new event introduces a new Juju context ID
+        # At this point the leader can take the lock
+        os.environ["JUJU_CONTEXT_ID"] = "new-context-id"
+        assert self.harness.charm.node_lock.acquired
+
+    @responses.activate
+    @patch("socket.socket.connect")
+    def test_node_lock_has_online_nodes_leader_acquire_lock_via_document(self, _):
+        # Initializing mocks showing the unit online in the cluster
+        mock_response_root(self.charm.unit.name, self.charm.opensearch.host)
+        mock_response_nodes(self.charm.unit.name, self.charm.opensearch.host)
+
+        # No unit has the lock at this moment
+        expected_response = {"unit-name": ""}
+
+        responses.add(
+            method="GET",
+            url=f"https://{self.charm.opensearch.host}:9200/.charm_node_lock/0",
+            json=expected_response,
+            status=200,
+        )
+
+        # The leader does not have the lock at this moment, but starts "queuing up"
+        # If the charm code raises an uncaught exception later in the Juju event
+        self.harness.set_leader(is_leader=True)
+        self.harness.update_relation_data(
+            self.lock_rel_id, f"{self.charm.app.name}", {"unit-with-lock": ""}
+        )
+        assert not self.harness.charm.node_lock.acquired
+
+        assert (
+            self.harness.get_relation_data(self.lock_rel_id, f"{self.charm.unit.name}")[
+                "lock-requested"
+            ]
+            == "true"
+        )
+
+        # A new event introduces a new Juju context ID
+        # At this point the leader can take the lock
+        os.environ["JUJU_CONTEXT_ID"] = "new-context-id"
+        assert self.harness.charm.node_lock.acquired
+
+    def test_node_lock_no_online_nodes_depating_node_doesnt_break(self):
```

Comment:
"departing"

Reply:
Thx much :-) 

Let me fix the typo in the "neighbor" PR (similar bugfix) (avoiding to loose 2 precious approvals ;-) )